src/lib/y2network/interfaces_collection.rb

```diff
@@ -57,6 +58,22 @@ def initialize(interfaces = [])
       @interfaces = interfaces
     end
 
+    # @param bridge_iface [Interface] an interface of bridge type
+    #
+    # @return [Array<Interface>] list of interfaces usable in bridge_iface
+    def select_bridgeable(bridge_iface)
+      raise "Bridge interface expected" unless bridge_iface.type == "br"
+      lan_items.select { |i| bridgeable?(bridge_iface, i) }
+    end
+
+    # @param bond_iface [Interface] an interface of bond type
+    #
+    # @return [Array<Interface>] list of interfaces usable in bridge_iface
+    def select_bondable(bond_iface)
+      raise "Bond interface expected" unless bond_iface.type == "bond"
+      lan_items.select { |i| bondable?(bond_iface, i) }
+    end
```

Comment:
For me this is very strange location. Why it is not method on Interface, VirtualInterface or ideally BondInterface class? Having it in collection where you need to pass name of element of collection is strange for me.

Reply:
BTW with this change object start looking for me quite big and also I start getting impression it can be soon another LanItems

Reply:
moved to the builder